src/Elastic.Apm/Metrics/MetricsCollector.cs

```diff
@@ -108,13 +130,18 @@ internal void CollectAllMetricsImpl()
 					if (samplesFromCurrentProvider != null)
 					{
 						var sampleArray = samplesFromCurrentProvider as MetricSample[] ?? samplesFromCurrentProvider.ToArray();
-						if (sampleArray.Any())
+						if (sampleArray.Length != 0 && sampleArray.All(x => !double.IsNaN(x.KeyValue.Value) && !double.IsInfinity(x.KeyValue.Value)))
```

Comment:
@SergeyKleyman, @gregkalapos, do we need to filter all samples from metrics provider as it did now in case of any value is either `NaN` or `Infinite`?

Reply:
>  do we need to filter all samples from metrics provider as it did now in case of any value is either NaN or Infinite?

Hmm, why don't we just filter out samples with `NaN` and `Infinity`? Maybe we can run into `IMetricsProvider` implementations where some numbers are wrong, but not all, in that case I'd still send the values that are correct numbers. No strong feeling though.

Reply:
Oh, I see, this would open up the question that if there is 1 `NaN` or `Infinite` in `sampleArray`, then is it a `ConsecutiveNumberOfFailedReads` or not, and how to handle that, right?

Reply:
Yeap, I thought about it and this is a reason why I asked it. So, filter out only invalid samples is reasonable for me. Will do changes